### PR TITLE
new CapabilitiesResolver implementation (ts only)

### PR DESCRIPTION
**Pull request by mmandlis**
not used yet, but backward compatibility checks are in place.

### CODE REVIEW

File: src/tools/storage-key-recipe-resolver.ts [hunk 1]
```diff
@@ -34,6 +37,7 @@ export class StorageKeyRecipeResolver {
 
   constructor(context: Manifest) {
     this.runtime = new Runtime({context});
+    // VolatileStorageKey.register();
```
Comment (csilvestrini) [inline src/tools/storage-key-recipe-resolver.ts:40]: Delete/uncomment?

File: src/runtime/tests/capabilities-resolver-test.ts [hunk 2]
```diff
@@ -37,7 +38,7 @@ describe('Capabilities Resolver', () => {
   const referenceType = new ReferenceType(entityType);
   const handleId = 'h0';
 
-  it('creates storage keys', Flags.withDefaultReferenceMode(async () => {
+    it('creates storage keys', Flags.withDefaultReferenceMode(async () => {
```
Comment (csilvestrini) [inline src/runtime/tests/capabilities-resolver-test.ts:41]: Indenting seems wrong

File: src/runtime/capabilities-new.ts [hunk 3]
```diff
@@ -263,127 +296,195 @@ export abstract class BooleanCapability extends Capability {
...
-  constructor(isQueryable: boolean) { super(isQueryable); }
-
-  static fromAnnotations(annotations: AnnotationRef[]): Capability {
+  static fromAnnotations(annotations: AnnotationRef[] = []): Capability {
     const queryableAnnotations =
         annotations.filter(annotation => annotation.name === 'queryable');
+    if (queryableAnnotations.length === 0) {
+      return null;
+    }
     assert(queryableAnnotations.length <= 1, `Containing multiple queryable annotations`);
```
Comment (csilvestrini) [inline src/runtime/capabilities-new.ts:305]: This assertion should probably be `=== 1` now

File: src/runtime/capabilities-new.ts [hunk 4]
```diff
@@ -263,127 +296,195 @@ export abstract class BooleanCapability extends Capability {
...
+// ramdisk and volatile memory. Currently only introduced for backward
+// compatibility in tests.
+export class Shareable extends BooleanCapability {
+  static fromAnnotations(annotations: AnnotationRef[] = []): Capability {
+    const shareableAnnotations =
+        annotations.filter(annotation => annotation.name === 'shareable');
+    if (shareableAnnotations.length === 0) {
+      return null;
+    }
+    assert(shareableAnnotations.length <= 1, `Containing multiple queryable annotations`);
```
Comment (csilvestrini) [inline src/runtime/capabilities-new.ts:326]: Here too, and in the rest

File: src/tools/storage-key-recipe-resolver.ts (hunk 1 above)
Comment (mmandlis) [reply src/tools/storage-key-recipe-resolver.ts]: removed

File: src/runtime/tests/capabilities-resolver-test.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/tests/capabilities-resolver-test.ts]: fixed

File: src/runtime/capabilities-new.ts (hunk 3 above)
Comment (mmandlis) [reply src/runtime/capabilities-new.ts]: done, thanks.

File: src/runtime/capabilities-new.ts [hunk 5]
```diff
@@ -108,10 +123,22 @@ export class Persistence extends Capability {
     return CapabilityComparison.LessStrict;
   }
 
+  toDebugString(): string {
+    switch (this.type) {
+      case PersistenceType.Unrestricted: return 'unrestrictedPersistence';
+      case PersistenceType.InMemory: return 'inMemory';
+      case PersistenceType.OnDisk: return 'onDisk';
+      case PersistenceType.None: return 'noPersistence';
```
[the remark this answers is not in the document]
Comment (mmandlis) [reply src/runtime/capabilities-new.ts]: done, thanks!

**Pull request merged**